Fix: Resolve undefined value issue in icon picker input field

RabbiIslamRony opened the pull request:
## PR Type
What kind of change does this PR introduce?
<!-- Mark completed items with an [x] -->
- [x] Bugfix
- [ ] Security fix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Text changes
- [ ] Other... Please describe:

## Description

### **Problem**
Icon picker input fields were displaying "undefined" as the value instead of showing proper placeholder text, creating a poor user experience.

### **Root Cause**
- Icon picker JavaScript was accepting the string "undefined" as a valid value
- No validation to filter out "undefined" string values
- Direct insertion of undefined values into HTML input fields

### **Solution**
✅ **Enhanced Value Validation:**
- Added check to reject "undefined" string values during initialization
- Implemented safe display value handling in renderMarkup function

### **Technical Changes**
```javascript
// Before (Problem)
this.value = (typeof args.value === 'string') ? args.value : this.value;
value="${this.value}"

// After (Solution)  
this.value = (typeof args.value === 'string' && args.value !== 'undefined') ? args.value : this.value;
let displayValue = (this.value && this.value !== 'undefined') ? this.value : '';
value="${displayValue}"
```

### **Files Modified**
- `assets/vendor-js/icon-picker/icon-picker.js`

Testing
✅ Icon picker input shows placeholder when no icon selected
✅ No "undefined" values appear in UI
✅ Proper icon selection and display functionality
✅ Reset functionality works correctly

## Any linked issues
Fixes #
https://taiga-sovware-u10698.vm.elestio.app/project/directorist/issue/2004?status=9,8,65,74,83,12,75&assigned_to=7,9&order_by=status

## Checklist

- [x] My code follows the [WordPress coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)

RabbiIslamRony committed:
Fix: Resolve undefined value issue in icon picker input field

- Add validation to reject 'undefined' string values
- Implement safe display value handling for input fields
- Prevent 'undefined' text from appearing in icon picker UI
- Improve user experience with proper placeholder display

iamibrahimriaz commented:
@RabbiIslamRony Great job handling the undefined value issue here — this fix will prevent display bugs and runtime errors. 👍

A small suggestion to make it even safer: escape the value before injecting it into the input, so any double quotes in this.value won’t break the HTML:

```
const displayValue = (this.value && this.value !== 'undefined') ? this.value : '';
const safeDisplayValue = String(displayValue).replace(/"/g, '&quot;');
```

Then use it in your markup:

`value="${safeDisplayValue}" style="${safeDisplayValue ? 'padding-left: 38px' : ''}"`

This keeps the project structure intact while improving HTML safety and maintainability.
Really solid work — love the attention to edge cases here! 👏

RabbiIslamRony committed:
Merge branch 'development' into fix/icon-picker-undefined-value-issue

RabbiIslamRony committed:
feat: improve HTML safety in icon picker by escaping values

- Add HTML escaping for displayValue to prevent injection attacks
- Replace double quotes with &quot; entity before injecting into HTML
- Maintain existing undefined value fix while adding security layer
- Use safeDisplayValue in span class, input value, and style attributes

iamibrahimriaz commented:
Looks Good
